### PR TITLE
node: add window properties (title, class, role)

### PR DESCRIPTION
**Pull request by rtgnx**
I have added window properties to node type.

### CODE REVIEW

**Comment by stapelberg** [comment]
Thanks for the pull request!

This package implements the documented i3 API, which currently does not include window_properties. Could you submit a pull request or file an issue over at https://github.com/i3/i3 to get the field documented? Afterwards, we can merge this PR. Thank you!

**Comment by Airblader** [comment]
Now merged in i3.

File: tree.go [hunk 1]
```diff
@@ -48,6 +48,12 @@ type Rect struct {
 	Height int64 `json:"height"`
 }
 
+type WindowProperties struct {
+	Title string `json:"title"`
+	Class string `json:"class"`
+	Role  string `json:"window_role"`
+}
```
Comment (stapelberg) [inline tree.go]: Could you add the remaining fields here which are now documented in i3 please?

File: tree.go [hunk 2]
```diff
@@ -48,6 +48,12 @@ type Rect struct {
 	Height int64 `json:"height"`
 }
 
+type WindowProperties struct {
```
Comment (stapelberg) [inline tree.go]: Can you add a comment on the type here please?

**Comment by rtgnx** [comment]
It's done now. Thanks for patience !

File: tree.go [hunk 3]
```diff
@@ -48,6 +48,15 @@ type Rect struct {
 	Height int64 `json:"height"`
 }
 
+// WindowProperties correspond to X11 window properties
```
Comment (stapelberg) [inline tree.go]: Could you add a link to https://build.i3wm.org/docs/ipc.html#_tree_reply here please? (Like in the comment on Node.)

Comment (rtgnx) [reply tree.go]: Comment updated :)

**Comment by stapelberg** [comment]
Thanks for the update! Unfortunately, the travis build fails. I think that’s unrelated to your change. I’ll take a look at that in a few minutes.

**Comment by stapelberg** [comment]
Sorry for the delay!

Fixed the travis issue with commit https://github.com/i3/go-i3/commit/75b22e34fa23911fccbdb2bc296000c168f16cd7, and finally merged your PR.

Thanks again for the contribution!

**Pull request merged**